devices/AtomQrCode/README.md

```diff
@@ -0,0 +1,86 @@
+# M5Stack ATOM QR Code reader
+
+This is a library to interact with the ATOM QR Code reader from M5Stack.
+
+![atom-qr-reader.png](./atom-qr-reader.png)
+
+## Documentation
+
+* The module documentation can be found [here](https://docs.m5stack.com/en/atom/atomic_qr)
+
+## Usage
+
+**Important**: make sure you properly setup the UART pins for ESP32 before creating the `QrCodeReader`. For this, make sure you install the `nanoFramework.Hardware.ESP32` NuGet:
+
+```csharp
+///////////////////////////////////////////////////////////////////////
+// when connecting to an ESP32 device, need to configure the UART GPIOs
+// the GPIOs below are the ones from the ATOM device that's part of the
+// ATOM QR Code reader kit from M5Stack.
+Configuration.SetPinFunction(22, DeviceFunction.COM2_RX);
+Configuration.SetPinFunction(19, DeviceFunction.COM2_TX);
+```
+
+For other devices, like STM32, please make sure you're using the pins for the UART you want to use.
+
+```csharp
+using Iot.Device.AtomQrCode;
+using nanoFramework.Hardware.Esp32;
+
+// reader is connected to COM2
+QrCodeReader reader = new QrCodeReader("COM2");
+```
+
+The reader offers various options to work with. Let's explore the most common usage scenarios.
+
+### Reading a barcode
+
+To perform readings "on-demand" the trigger mode has to be set to `Host` and then calling `TryReadBarcode()`.
+
+```csharp
+// set scan mode to HOST so it reads on request
+reader.TriggerMode = TriggerMode.Host;
+
+// try reading barcode
+string code;
+reader.TryReadBarcode(out code);
```

Comment:
maybe make it with a if() and else and print the code if successful, otherwise print error so it's clear how to use the pattern